model/model-assembler/src/main/proto/spine/model/spine_model.proto

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright 2017, TeamDev Ltd. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+syntax = "proto3";
+
+package spine.model;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_generate_equals_and_hash = false;
+option java_multiple_files = true;
+option java_outer_classname = "SpineModelProto";
+option java_package = "io.spine.model";
+
+// The Spine type model.
+//
+// This message represents raw (i.e. non-typed) model of a Spine project.
+//
+// The message contains the type names of the Spine core entities found in a Spine project (not
+// necessarily grouped by the entity kind).
+//
+message SpineModel {
```

Comment:
Trying to look objectively it's really difficult to explain why this message is called the same as `io.spine.server.model.Model`, but has a weird structure.

That's why I would rename this message to `CommandHandlers`. And reduce the reserved field count to like `2 to 5`.